scripts/prepare.sh

```diff
@@ -30,9 +35,44 @@ if [[ "${WORKSPACES_SSH_DIRECTORY_BOUND}" == "true" ]]; then
 	chmod -Rv 400 ${dot_ssh}/*
 fi
 
-if [[ "${PARAMS_DELETE_EXISTING}" == "true" ]]; then
-	phase "Deleting all contents of checkout-dir '${checkout_dir}'"
-	clean_dir ${checkout_dir} || true
+# Setting up the config for the git.
+
+if [ -n "${PARAMS_GIT_USER_EMAIL}" ] ; then
+	echo "${[ -n "${PARAMS_GIT_USER_EMAIL}"]}"
+    #phase "Setting global email for git ${PARAMS_GIT_USER_EMAIL}"
+    git config --global user.email "${PARAMS_GIT_USER_EMAIL}"
 fi
 
+if [ -n "${PARAMS_GIT_USER_NAME}" ] ; then
+    #phase "Setting global username for git ${PARAMS_GIT_USER_NAME}"
+    git config --global user.name "${PARAMS_GIT_USER_NAME}"
+fi
+
+#
+# CA (`ssl-ca-directory` Workspace)
+#
+
+if [[ "${WORKSPACES_SSL_CA_DIRECTORY_BOUND}" == "true" && -n "${PARAMS_CRT_FILENAME}" ]]; then
+	phase "Inspecting 'ssl-ca-directory' workspace looking for '${PARAMS_CRT_FILENAME}' file"
+	crt="${WORKSPACES_SSL_CA_DIRECTORY_PATH}/${PARAMS_CRT_FILENAME}"
+	[[ ! -f "${crt}" ]] &&
+		fail "CRT file (PARAMS_CRT_FILENAME) not found at '${crt}'"
+
+	phase "Exporting custom CA certificate 'GIT_SSL_CAINFO=${crt}'"
+	export GIT_SSL_CAINFO=${crt}
+fi
+
+#
+# Proxy Settings
+#
+
+phase "Setting up HTTP_PROXY='${PARAMS_HTTP_PROXY}'"
+[[ -n "${PARAMS_HTTP_PROXY}" ]] && export HTTP_PROXY="${PARAMS_HTTP_PROXY}"
+
+phase "Settting up HTTPS_PROXY='${PARAMS_HTTPS_PROXY}'"
+[[ -n "${PARAMS_HTTPS_PROXY}" ]] && export HTTPS_PROXY="${PARAMS_HTTPS_PROXY}"
+
+phase "Setting up NO_PROXY='${PARAMS_NO_PROXY}'"
+[[ -n "${PARAMS_NO_PROXY}" ]] && export NO_PROXY="${PARAMS_NO_PROXY}"
+
```

Comment:
Given the Task is split into three steps ([`prepare`, `git-clone` and `report`](https://github.com/openshift-pipelines/task-git/blob/6c85da21eae3f29bd80869ff5d97006d7e0c254e/templates/task.yaml#L166-L192)), the exported variables won't be passed from `prepare` to `git-clone`. And, that's why the `export` settings were bundled on `git-clone.sh` before.

I think the best course of action is to make sure any `export` variable is performed right before `git-init` (or an arbitrary `git` script).